model/ArrayList.php

```diff
@@ -422,10 +422,15 @@ public function sort() {
 			throw new InvalidArgumentException("Bad arguments passed to sort()");
 		}
 
+		// Store the original positions of the items as a fallback, so we can preserve the original order in the event
+		// that array_multisort is unable to work out a sort order for them. This also prevents array_multisort trying
+		// to inspect object properties which can result in errors with circular dependencies
+		$originalPositions = (count($this->items)) ? range(1, count($this->items)) : array();
```

Comment:
would `array_keys()` not work just as well for this (assuming it's not an associative array)?